atlasexec/atlas.go

```diff
@@ -319,21 +319,35 @@ func (c *Client) MigrateLint(ctx context.Context, params *MigrateLintParams) (*S
 		return nil, errors.New("atlasexec: Writer or Web reporting are not supported with MigrateLint, use MigrateLintError")
 	}
 	r, err := c.runCommand(ctx, lintArgs(params))
+	var cliErr cliError
+	ok := errors.As(err, &cliErr)
+	if ok && cliErr.hasLintErrors() {
+		r = strings.NewReader(cliErr.detail)
+		err = nil
+	}
```

Comment:
Please see this: https://github.com/ariga/atlas/blob/master/cmd/atlas/internal/cmdapi/cmdapi_oss.go#L101-L110